git_theta/updates/dense.py

```diff
@@ -0,0 +1,92 @@
+"""Class managing dense parameter updates."""
+
+import logging
+import os
+import shutil
+from typing import Optional
+import numpy as np
+from git_theta import file_io
+from git_theta import params
+from git_theta.updates import Update, UpdateConstants
+
+
+class DenseUpdate(Update):
+    """An update where all parameters are changed."""
+
+    @property
+    def name(self):
+        """The name used to get this class as a plugin."""
+        return "dense"
+
+    def read(self, path: str) -> np.ndarray:
+        """Read the parameter values in the path dir.
+
+        Note:
+            In this case, the read parameter values are the whole update, all
+            values are overwritten so the result of this read will just be
+            returned.
+
+        Parameters
+        ----------
+        path
+            The .../params/updates/${hash} directory path that contains parameter
+            update values.
+
+        Returns
+        -------
+        np.ndarray
+            The parameter values at path.
+        """
+        logging.debug(f"Reading Dense Update from '{path}'.")
+        return file_io.load_tracked_file(path)
+
+    def write(self, path: str, parameter: np.ndarray):
+        """Write `parameter` values to path.
+
+        Note:
+            This update type removes all the previous updates in the .../updates/
+            directory as they are no longer needed. This update sets all parameter
+            values.
+
+        Parameters
+        ----------
+        path
+            The .../params/updates/${hash} directory path to write to.
+        parameter
+            The values to write.
+        """
+        # Remove any past updates to begin anew.
+        shutil.rmtree(path)
```

Comment:
Should this be `git_utils.remove_file`? Not sure if git will stage the previous contents of `updates/` to be removed after `shutil.rmtree`.

Reply:
The root dir of all this parameter stuff (`path` here, i.e. `.../param/`) is eventually the target of a git add. That add on the dir will stage the deletion of all the contents of `.../param/updates` as well as the addition of the new values.

Can the remove file function handle directories?

Reply:
That makes sense. `remove_file` (bad name) will run `git rm -r` if the argument is a directory. Right now the way it's called in `git theta add` it gets called on the `params/` directory.

Reply:
Currently using `remove_file` doesn't handle the case where there is no tracked file well (see https://github.com/r-three/git-theta/issues/104) so I added a TODO to switch to it from shutil once it handles that edges case